### DISCUSSION TITLE
Extracting specific data from EnableOutput in OR-Tools every 10 seconds

### DISCUSSION THREAD

**Discussion by gerymueller** [google/or-tools#3900] [General]
Hello OR-Tools community,

I've been using OR-Tools with the SCIP solver, specifically using the following setup:

```
self.solver = pywraplp.Solver.CreateSolver('SCIP')
self.solver.EnableOutput()
status = self.solver.Solve()
```

While the solver is running, the EnableOutput method provides a detailed console output with a lot of useful information, like this:

```
time | node  | left  |LP iter|LP it/n|mem/heur|mdpt |vars |cons |rows |cuts |sepa|confs|strbr|  dualbound   | primalbound  |  gap   | compl. 
111s|  8200 |   131 |980573 |  83.9 |    63M |  69 | 693 |1489 |1195 |6281 |  0 |  13k|5395 | 1.302965e+04 | 1.401400e+04 |   7.55%|  77.77%
112s|  8300 |    60 |992733 |  84.2 |    63M |  69 | 693 |1498 | 801 |6443 |  1 |  13k|5446 | 1.317840e+04 | 1.401400e+04 |   6.34%|  91.13%
```

For my specific use-case, I'd like to capture specific data points from this output, say, the gap, every 10 seconds while the solver is running. Is there a built-in way within OR-Tools to extract this data? Alternatively, I'm open to monitoring and parsing the console output using external tools or methods if that's a recommended approach.

Any guidance, recommendations, or workarounds would be much appreciated.

Thank you in advance!

**Comment by sschnug** [comment]
I doubt that there is much prebuilt functionality. MP wrappers are relatively thin imho.

During solving? That's much harder than after solving.

If the latter is enough, just use a python script or similar and parse it. I have done that in the past with some more verbose SCIP output (w/o ortools) as getting access to some stats @ scip internally is surprisingly annoying.

If you really need it during solving, i would prefer setting up a callback to obtain improved incumbents and calc the gap myself before i would redirect streams and parse stuff.
This should be possible, but i never used the scip mp-wrapper and it's callback abstractions.
